config/configtls/configtls.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configtls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// TLSConfig exposes the common client and server TLS configurations.
+// Note: Since there isn't anything specific to a server connection. Components
+// with server connections should use TLSConfig.
+type TLSConfig struct {
+	// Path to the CA cert. For a client this verifies the server certificate.
+	// For a server this verifies client certificates. If empty uses system root CA.
+	// (optional)
+	CAFile string `mapstructure:"ca_file"`
+	// Path to the TLS cert to use for TLS required connections. (optional)
+	CertFile string `mapstructure:"cert_file"`
+	// Path to the TLS key to use for TLS required connections. (optional)
+	KeyFile string `mapstructure:"key_file"`
+}
+
+// TLSClientConfig contains TLS configurations that are specific to client
+// connections in addition to the common configurations. This should be used by
+// components configuring TLS client connections.
+type TLSClientConfig struct {
+	TLSConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// These are config options specific to client connections
+	// In gRPC when set to true, this is used to disable the client transport security.
+	// See https://godoc.org/google.golang.org/grpc#WithInsecure.
+	// In HTTP, this disables verifying the server's certificate chain and host name
+	// (InsecureSkipVerify in the tls Config).
+	// This is only used for client connections. (optional, default false)
+	UseInsecure bool `mapstructure:"insecure"`
```

Comment:
nit: Would suggest just calling this `Insecure` like the config name, I don't think the `Use` adds any extra information.